lib/server/services/run/RunService.js

```diff
@@ -318,6 +328,23 @@ const updateEorReasonsOnRun = async (runId, userName, eorReasonPatches) => {
     await EorReasonRepository.addMany(newEorReasons);
 };
 
+/**
+ * Method to process a user for the run
+ *
+ * @param {User|null} userData - user object, either userO2Start or userO2Stop
+ * @param {number} [userData.externalId] - external identifier of the user
+ * @param {string|null} [userData.name] - name of the user or empty
+ * @returns {Promise<number|null>} - returns the id of the user or null if the userData is null
+ */
+const processUser = async (userData) => {
```

Comment:
I would rather name this function `getUserIdFromExternalId` and take as parameter the external id directly.

Reply:
done

Reply:
I apologize, I read to quickly.
I would rather name it getOrCreateUser, take the userData as you did, and return the id or null.
The you simply have something like
```js
newRun.userIdO2Start = (await getOrCreateUser(userData))?.id
```

Reply:
No problem, I changed it now.